docs/import/add_disaster.js

```diff
@@ -0,0 +1,289 @@
+    $('#selected-disaster').show();
+    const assetPickerDiv = $('#asset-pickers');
+    // We are already doing this inside createAssetPickers but not until
+    // after the first promise completes so also do it here so lingering
+    // pickers from previous disasters don't hang around.
+    assetPickerDiv.empty();
+    return assetPickersDone;
+  }
+}
+
+/**
+ * Writes the given details to a new disaster entry in firestore. Fails if
+ * there is an existing disaster with the same details.
+ * @param {string} disasterId of the form <year>-<name>
+ * @param {Array<string>} states array of state (abbreviations)
+ * @return {Promise<boolean>} returns true after successful write to firestore.
+ */
+function writeNewDisaster(disasterId, states) {
+  if (disasters.has(disasterId)) {
+    setStatus('Error: disaster with that name and year already exists.');
+    return Promise.resolve(false);
+  }
+
+  disasters.set(disasterId, states);
+  toggleState(disasterId);
+
+  clearStatus();
+  const disasterOptions = $('#disaster > option');
+  // comment needed to quiet eslint on no-invalid-this rules
+  disasterOptions.each(/* @this HTMLElement */ function() {
+    if ($(this).val() > disasterId) {
+      $(createOptionFrom(disasterId)).insertBefore($(this));
+      return false;
+    }
+  });
+  $('#disaster').val(disasterId);
+  return getFirestoreRoot()
+      .collection('disaster-metadata')
+      .doc(disasterId)
+      .set({states: states})
+      .then(() => true);
+}
+
+/**
+ * Onclick function for the new disaster form. Writes new disaster to firestore,
+ * local disasters map and disaster picker. Doesn't allow name, year or states
+ * to be empty fields.
+ * @return {Promise<boolean>} resolves true if new disaster was successfully
+ *     written.
+ */
+function addDisaster() {
+  const year = $('#year').val();
+  const name = $('#name').val();
+  const states = $('#states').val();
+
+  if (!year || !name || !states) {
+    setStatus('Error: Disaster name, year, and states are required.');
+    return Promise.resolve(false);
+  }
+  if (isNaN(year)) {
+    setStatus('Error: Year must be a number.');
+    return Promise.resolve(false);
+  }
+  if (notAllLowercase(name)) {
+    setStatus(
+        'Error: disaster name must be comprised of only lowercase letters');
+    return Promise.resolve(false);
+  }
+  const disasterId = year + '-' + name;
+  return writeNewDisaster(disasterId, states);
+}
+
+/**
+ * Returns true if the given string is *not* all lowercase letters.
+ * @param {string} val
+ * @return {boolean}
+ */
+function notAllLowercase(val) {
+  return !/^[a-z]+$/.test(val);
+}
+
+// Needed for testing :/
+const emptyCallback = () => {};
+
+// TODO: add functionality for adding assets to disaster records from these
+// pickers.
+/**
+ * Requests all assets in ee directories corresponding to given states.
+ * @param {Array<string>} states e.g. ['WA']
+ * @return {Promise<Array<string | Array<string>>>} 2-d array of all retrieved
```

Comment:
nit: I think this signature should be Promise<Array<Array<string, Array<string>>>>, right? Because it's an array (one element per state), and each element of that array is itself a 2-element array, first element state, and second element array of strings.